perception_dataset/t4_dataset/annotation_files_generator.py

```diff
@@ -90,24 +99,27 @@ def convert_one_scene(
             print(e)
 
         nuim = NuImages(version="annotation", dataroot=input_dir, verbose=False)
-        frame_index_to_sample_data_token: List[Dict[int, str]] = [{} for x in range(6)]
-        mask: List[Dict[int, str]] = [{} for x in range(6)]
+        # FIXME: Avoid hard coding the number of cameras
+        num_cameras = 6 if self._camera2idx is None else len(self._camera2idx)
+        frame_index_to_sample_data_token: List[Dict[int, str]] = [{} for _ in range(num_cameras)]
+        mask: List[Dict[int, str]] = [{} for x in range(num_cameras)]
 
         has_2d_annotation: bool = False
         for frame_index in sorted(scene_anno_dict.keys()):
             anno_list: List[Dict[str, Any]] = scene_anno_dict[frame_index]
             for anno in anno_list:
-                if "two_d_box" in anno.keys():
+                if "two_d_box" in anno.keys() or "two_d_segmentation" in anno.keys():
                     has_2d_annotation = True
                     break
 
         if has_2d_annotation:
+            # NOTE: num_cameras is always 6, because it is hard coded above.
             for frame_index_nuim, sample_nuim in enumerate(nuim.sample_data):
                 if (
                     sample_nuim["fileformat"] == "png" or sample_nuim["fileformat"] == "jpg"
                 ) and sample_nuim["is_key_frame"]:
                     cam = sample_nuim["filename"].split("/")[1]
-                    cam_idx = self._camera2idx[cam]
+                    cam_idx = self._camera2idx[cam] - 1
```

Comment:
```suggestion
                    cam_idx = self._camera2idx[cam]
```

[must]

We need to remove `-1` if we add [this commit](https://github.com/tier4/tier4_perception_dataset/pull/164/commits/181947daa583c7facdee369f4112055a331ab4e5)

```bash
-> frame_index_to_sample_data_token[cam_idx].update(
(Pdb) cam_idx
-1
```

Reply:
Addressed in 2e2aeeb822ca87a566be08084c8fdb5ddd06d21b